app/services/plugin.service.ts

```diff
@@ -193,6 +155,42 @@ export class PluginService {
         }
     }
 
+    public loadPluginTypeDefinitions(src: string): Definitions {
+
+        const scope = Type.parseScope(src);
+        scope.validate();
+
+        const nodes: [string, Type.ClassType][] = [];
+        const edges: [string, Type.ClassType][] = [];
+        const graphs: [string, Type.ClassType][] = [];
+
+        for (const [name, value] of scope.definitions.entries()) {
+            if (value.subtype(Type.Node)) {
+                nodes.push([name, value as Type.ClassType]);
+            }
+            if (value.subtype(Type.Edge)) {
```

Comment:
It might be a good idea then to put in a comment stating as such so that no one changes it to an else if block.